blog/enable-cilium/index.mdx

```diff
@@ -0,0 +1,477 @@
+DaemonSet              cilium             Desired: 1, Ready: 1/1, Available: 1/1
+Containers:            cilium             Running: 1
+                       cilium-operator    Running: 1
+Cluster Pods:          1/2 managed by Cilium
+Helm chart version:
+Image versions         cilium             quay.io/cilium/cilium:v1.15.5@sha256:4ce1666a73815101ec9a4d360af6c5b7f1193ab00d89b7124f8505dee147ca40: 1
+                       cilium-operator    quay.io/cilium/operator-generic:v1.15.5@sha256:f5d3d19754074ca052be6aac5d1ffb1de1eb5f2d947222b5f10f6d97ad4383e8: 1
+```
+
+Add `nodeAffinity` for Cilium `DaemonSet` to make sure these pods are only created on cloud nodes.
+These Cilium pods are generic Cilium `DaemonSet`, so supposed to be running on cloud nodes but where `EdgeCore` running.
+
+```
+### Edit Cilium DaemonSet with the following patch
+>  kubectl edit ds -n kube-system cilium
+```
+
+```patch
+diff --git a/cilium-kubelet.yaml b/cilium-kubelet.yaml
+index 21881e1..9946be9 100644
+--- a/cilium-kubelet.yaml
++++ b/cilium-kubelet.yaml
+@@ -29,6 +29,12 @@ spec:
+         k8s-app: cilium
+     spec:
+       affinity:
++        nodeAffinity:
++          requiredDuringSchedulingIgnoredDuringExecution:
++            nodeSelectorTerms:
++              - matchExpressions:
++                - key: node-role.kubernetes.io/edge
++                  operator: DoesNotExist
+         podAntiAffinity:
+           requiredDuringSchedulingIgnoredDuringExecution:
+           - labelSelector:
+```
+
+After editing, Cilium pods will be restarted.
+
+<!--truncate-->
+
+### KubeEdge CloudCore Setup
+
+1st of all, we need to install `Keadm` with the official procedure [Installing KubeEdge with Keadm](https://kubeedge.io/docs/setup/install-with-keadm).
+
+In this blog, we use `Keadm v1.16.1` as following.
+
+```bash
+### Install v1.16.1 keadm command
+> wget https://github.com/kubeedge/kubeedge/releases/download/v1.16.1/keadm-v1.16.1-linux-amd64.tar.gz
+> tar -zxvf keadm-v1.16.1-linux-amd64.tar.gz
+> cp keadm-v1.16.1-linux-amd64/keadm/keadm /usr/local/bin
+
+> keadm version
+version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.1", GitCommit:"bd7b42acbfbe3a453c7bb75a6bb8f1e8b3db7415", GitTreeState:"clean", BuildDate:"2024-03-27T02:57:08Z", GoVersion:"go1.20.10", Compiler:"gc", Platform:"linux/amd64"}
+```
+
+and then, start the `CloudCore` with `v1.16.1`.
+
+```
+> keadm init --advertise-address="AA.BBB.CCC.DD" --profile version=v1.16.1 --kube-config=/root/.kube/config
+Kubernetes version verification passed, KubeEdge installation will start...
+CLOUDCORE started
+=========CHART DETAILS=======
+NAME: cloudcore
+LAST DEPLOYED: Tue Jun  4 17:19:15 2024
+NAMESPACE: kubeedge
+STATUS: deployed
+REVISION: 1
+```
+
+After `CloudCore` is started, we also need to enable `DynamicControllers`.
+
+```
+### edit ConfigMap of CloudCore to enable dynamicController
+> kubectl edit cm -n kubeedge cloudcore
+> kubectl rollout restart deployment -n kubeedge cloudcore
```

Comment:
this operation does not work; the pod should be deleted manually.

```sh
ultraman@ultraman-desktop:~$ sudo kubectl get pod -A -o wide                                                                                                         
NAMESPACE     NAME                                       READY   STATUS    RESTARTS   AGE     IP              NODE               NOMINATED NODE   READINESS GATES           
kubeedge      cloudcore-7bfc66cfbf-zzw95                 0/1     Pending   0          39s     <none>          <none>             <none>           <none>             
kubeedge      cloudcore-854cccfd6c-pqfhw                 1/1     Running   0          61s     43.82.125.157   ultraman-desktop   <none>           <none>             
ultraman@ultraman-desktop:~$ sudo kubectl describe pod -n kubeedge cloudcore-7bfc66cfbf-zzw95 
Events:
  Type     Reason            Age   From               Message
  ----     ------            ----  ----               -------
  Warning  FailedScheduling  40s   default-scheduler  0/1 nodes are available: 1 node(s) didn't have free ports for the requested pod ports. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
```

`sudo kubectl delete pod -n kubeedge  cloudcore-854cccfd6c-pqfhw`

Reply:
right, i did recheck this command.

it should be the following instead,

```shell
kubectl rollout restart deployment/cloudcore -n kubeedge
```

but technically we do not need `rollout` here, we can just delete the pod as suggested.

Reply:
i would suggest `kubectl delete pod -n kubeedge --selector=kubeedge=cloudcore` so that we do not need postfix strings.